src/main/java/se/sundsvall/mex/businesslogic/worker/AbstractWorker.java

```diff
@@ -1,25 +1,68 @@
 package se.sundsvall.mex.businesslogic.worker;
 
+import static se.sundsvall.mex.Constants.CAMUNDA_VARIABLE_CASE_NUMBER;
+import static se.sundsvall.mex.Constants.CAMUNDA_VARIABLE_REQUEST_ID;
 import static se.sundsvall.mex.Constants.FALSE;
 import static se.sundsvall.mex.Constants.UPDATE_AVAILABLE;
 
 import org.camunda.bpm.client.task.ExternalTask;
+import org.camunda.bpm.client.task.ExternalTaskHandler;
+import org.camunda.bpm.client.task.ExternalTaskService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import generated.se.sundsvall.camunda.VariableValueDto;
+import generated.se.sundsvall.casedata.ErrandDTO;
+import se.sundsvall.dept44.requestid.RequestId;
+import se.sundsvall.mex.businesslogic.handler.FailureHandler;
 import se.sundsvall.mex.integration.camunda.CamundaClient;
+import se.sundsvall.mex.integration.casedata.CaseDataClient;
 
-abstract class AbstractWorker {
+public abstract class AbstractWorker implements ExternalTaskHandler {
+	private final Logger logger;
 
 	private final CamundaClient camundaClient;
 
-	protected AbstractWorker(CamundaClient camundaClient) {
+	protected final CaseDataClient caseDataClient;
+
+	protected final FailureHandler failureHandler;
+
+	protected AbstractWorker(CamundaClient camundaClient, CaseDataClient caseDataClient, FailureHandler failureHandler) {
+		this.logger = LoggerFactory.getLogger(getClass());
 		this.camundaClient = camundaClient;
+		this.caseDataClient = caseDataClient;
+		this.failureHandler = failureHandler;
 	}
 
 	protected void clearUpdateAvailable(ExternalTask externalTask) {
 		/*
 		 * Clearing process variable has to be a blocking operation.
 		 * Using ExternalTaskService.setVariables() will not work without creating race conditions.
 		 */
-		camundaClient.setProcessInstanceVariable(externalTask.getProcessInstanceId(), UPDATE_AVAILABLE, FALSE);
+		setProcessInstanceVariable(externalTask, UPDATE_AVAILABLE, FALSE);
+	}
+
+	protected void setProcessInstanceVariable(ExternalTask externalTask, String variableName, VariableValueDto variableValue) {
+		camundaClient.setProcessInstanceVariable(externalTask.getProcessInstanceId(), variableName, variableValue);
+	}
+
+	protected ErrandDTO getErrand(ExternalTask externalTask) {
+		return caseDataClient.getErrandById(externalTask.getVariable(CAMUNDA_VARIABLE_CASE_NUMBER));
+	}
+
+	protected void logInfo(String msg, Object... arguments) {
```

Comment:
So in some of the worker classes that extend from this you are using this logInfo and logException methods and in some of the workers you are setting up a logger in the class. Decide on one way and use that.